internal/controller/jwt/vault.go

```diff
@@ -40,6 +40,25 @@ type Role struct {
 	TokenType            string                 `json:"token_type"`
 }
 
+// Validate validates if a role follow vault constraints
+// Check https://developer.hashicorp.com/vault/api-docs/auth/jwt#create-role to see vault contraints for JWT/OIDC roles
+func (role *Role) Validate() error {
+	if role.RoleType == "jwt" && len(role.BoundAudiences) == 0 {
```

Comment:
As the first part of each `if` statement are the same for all validations, we can refactor them, example:
```
if role.RoleType == "jwt" {
    if len(...) == 0 {
      return error
   }
   if...
   if...

}
```